File: 3206-find-common-elements-between-two-arrays/solution.py
class Solution:
    def findIntersectionValues(self, nums1: List[int], nums2: List[int]) -> List[int]:
        n1=0
        n2=0
        for i in nums1:
            if i in nums2:
                n1+=1
        for i in nums2:
            if i in nums1:
                n2+=1
        l=[]
        l.append(n1)
        l.append(n2)
        return l
        

File: 3206-find-common-elements-between-two-arrays/solution.java
class Solution {
    public int[] findIntersectionValues(int[] nums1, int[] nums2) {
        HashSet<Integer> setNums = new HashSet<>();
        for(int n : nums2) setNums.add(n);

        int c = 0;
        for(int n : nums1) {
            if(setNums.contains(n)) {
                c++;
            }
        }
        HashSet<Integer> setNums1 = new HashSet<>();
        for(int n : nums1) setNums1.add(n);

        int c1 = 0;
        for(int n : nums2) {
            if(setNums1.contains(n)) {
                c1++;
            }
        }
        return new int[]{c,c1};
    }
}
